includes/MediaWiki/Request/EditRequestSaver.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\Request;
+
+use MediaWiki\Extension\WikibaseReconcileEdit\Reconciliation\ItemReconciler;
+use MediaWiki\Extension\WikibaseReconcileEdit\Reconciliation\ReconciledItem;
+use MediaWiki\Extension\WikibaseReconcileEdit\Reconciliation\ReconciliationServiceItem;
+use Status;
+use User;
+use Wikibase\Repo\EditEntity\MediawikiEditEntityFactory;
+
+/**
+ * Service for saving entities
+ *
+ * @license GPL-2.0-or-later
+ */
+class EditRequestSaver {
+
+	/** @var MediawikiEditEntityFactory */
+	protected $editEntityFactory;
+
+	/** @var ItemReconciler */
+	protected $itemReconciler;
+
+	public function __construct(
+		MediawikiEditEntityFactory $editEntityFactory,
+		ItemReconciler $itemReconciler
+	) {
+		$this->editEntityFactory = $editEntityFactory;
+		$this->itemReconciler = $itemReconciler;
+	}
+
+	/**
+	 * @param EditRequest[] $requests
+	 * @throws \MediaWiki\Extension\WikibaseReconcileEdit\ReconciliationException
+	 */
+	public function persistEdits( array $requests, string $editToken, User $user ): Status {
+		$status = Status::newGood( [] );
+		foreach ( $requests as $request ) {
+			$reconciledItem = $this->itemReconciler->reconcileItem(
+				$request->entity(),
+				$request->reconcilePropertyId()
+			);
+			$otherItems = $request->otherItems();
+
+			$otherStatus = $this->persistItem( $reconciledItem, $otherItems, $editToken, $user );
+			if ( !$otherStatus->isOk() ) {
+				break;
+			}
+			$status->setResult( true, array_merge( $status->getValue(), [ $otherStatus->getValue() ] ) );
+			$status->merge( $otherStatus );
+		}
+
+		return $status;
+	}
+
+	/**
+	 * @param ReconciledItem $reconciledItem
+	 * @param ReconciliationServiceItem[] $otherItems
```

Comment:
The `$editToken` and `$user` params seem to be missing.

Reply:
done